app/models/text_field_type.rb

```diff
@@ -25,14 +28,23 @@ def mapping_field_name
   end
 
   def text_present
-    errors.add(:text, "must be present") if @text.empty?
+    errors.add(:text, 'must be present') if @text.empty?
   end
 
   def text_length
     validator = LengthValidator.new(validations[:length].merge(attributes: [:text]))
     validator.validate_each(self, :text, text)
   end
 
+  def text_unique
+    errors.add(:base, 'Slug Must be unique') unless FieldItem.jsonb_contains(:data, text: text).empty?
+    true
+  end
+
+  def validate_uniqueness?
```

Comment:
Alongside the ActiveModel comments, I would say let's go ahead and extract this out to the general FieldType model like we do with the other basic Validations. While it's true we don't use it right now the possibility of using it in the future *does* exist, and if we're building this with open source in mind we should adhere to that